### PR TITLE
first pass store unary

### PR DESCRIPTION
**Pull request by GertjanBisschop**
First pass at adding an option to keep unary nodes (#2128). Unit tests are very basic at the moment. 
I went with the option to explicitly pass the `parent_node_id` to `store_arg_edges()`. This seems to make sense going forward with different options of `keep_unary` but might be wrong here. 
Also, I stuck with the `store_full_arg`, `keep_full_arg` naming conventions for  now.

### CODE REVIEW

File: tests/test_algorithms.py [hunk 1]
```diff
@@ -34,6 +34,26 @@ def has_discrete_genome(ts):
...
+        for i in range(ts.num_nodes):
+            n = tree.num_children_array[i]
+            if n > 0:
+                if min_children[i] > 0:
+                    min_children[i] = min(min_children[i], n)
+                else:
+                    min_children[i] = n
+                max_children[i] = max(max_children[i], n)
+
+    assert any(n == 1 for n in min_children)
```
Comment (jeromekelleher) [inline tests/test_algorithms.py:51]: ``assert np.any(min_children == 1)`` here is a bit more efficient

File: tests/test_ancestry.py [hunk 2]
```diff
@@ -228,6 +228,36 @@ def test_gene_conversion_n100(self):
...
+            for i in range(ts.num_nodes):
+                n = tree.num_children_array[i]
+                if n > 0:
+                    if min_children[i] > 0:
+                        min_children[i] = min(min_children[i], n)
+                    else:
+                        min_children[i] = n
+                    max_children[i] = max(max_children[i], n)
+
+        assert any(n == 1 for n in min_children)
```
Comment (jeromekelleher) [inline tests/test_ancestry.py:255]: Ditto here

**Review by jeromekelleher — APPROVED** [review]
LGTM! Let's merge this and then figure out how to this intersects with things like simulating through the pedigree etc

**Comment by jeromekelleher** [comment]
Can merge after #2131 is resolved

**Comment by jeromekelleher** [comment]
@mergifyio rebase

**Comment by petrelharp** [comment]
Just checking - is this storing *all* the unary nodes?

**Comment by jeromekelleher** [comment]
This is "store the full extent of edges recording the paths of ancestry through nodes that are coalescent somewhere along their span". There's no recombination nodes - we need to think through the API before release to make something that's not too confusing (#2132)

**Pull request merged**